core/templates/services/math-interactions.service.ts

```diff
@@ -524,6 +524,42 @@ export class MathInteractionsService {
   // The input terms to this function should be the terms split by '+'/'-'
   // from an expression.
   termsMatch(term1: string, term2: string): boolean {
+    let getConstants = function(expression: string): string[] {
+      let constants = expression.match(/(\d*\.\d+)|(\d+)/g);
+      if (constants !== null) {
+        return constants;
+      }
+      return [];
+    };
+
+    let partsMatch = function(part1: string, part2: string): boolean {
+      // Nerdamer simplifies constant terms automatically so to avoid that,
+      // we'll replace all constants with a variable.
```

Comment:
Explain whether multiple instances of the same numeric value are replaced with different variables or not.

Reply:
Done.